pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -337,6 +337,17 @@ in rec {
         validity.handled
         ({
            overrideAttrs = f: mkDerivation (attrs // (f attrs));
+           # A derivation that always builds successfully and whose runtime
+           # dependencies are the original derivations build time dependencies
+           # This allows easy building and distributing of all derivations
+           # needed to enter a nix-shell with
+           #   nix-build shell.nix -A inputDerivation
+           inputDerivation = derivation (derivationArg // {
+             _derivation_original_builder = derivationArg.builder;
+             _derivation_original_args = derivationArg.args;
+             builder = stdenv.shell;
+             args = [ "-c" " export > $out" ];
```

Comment:
might be nice for those reading the code-base/pr to explain this line.

**export** or **env** (my preferred) when called with 0 arguments dumps out all current environment variables.
The end result of this derivation is a file whose contents are the environment variables.
_nix-shell_ works mainly by placing all _buildInputs_ into the subshell's _$PATH_ variable.
They are therefore present in plaintext in the file and can then be found by nix when grepping for runtime dependencies.

Reply:
Good point! I chose `export` because it's a bash builtin. I added an explanation of the how it works

Reply:
:+1: 
I'm still somewhat new to Nix ~6months (non-NixOS user); and a big fan of comments with rationale :)